streamer/pulp/streamer/server.py

```diff
@@ -129,7 +159,7 @@ def _handle_get(self, request):
                 for entry in q_set.all():
                     logger.info('Trying URL: {url}'.format(url=entry.url))
                     try:
-                        last_report = self._download(entry, responder)
+                        last_report = self._download(request, entry, responder)
```

Comment:
So this is a request that was made by some external client, and is being handled by the streamer, right?

What's unintuitive to me is that this object gets passed down to the download listener, which then sets headers on this request object. Am I understanding this right? Why does it make sense to set a header on a request that's already been made, and is currently being handled?